includes/Modules/AdSense.php

```diff
@@ -538,8 +538,16 @@ protected function create_adsense_earning_data_request( array $args = array() )
 		}
 
 		$host = wp_parse_url( $this->context->get_reference_site_url(), PHP_URL_HOST );
+
 		if ( ! empty( $host ) ) {
-			$opt_params['filter'] = 'DOMAIN_NAME==' . $host;
+			if ( strpos( $host, 'www.' ) !== false ) {
+				$host_with_www    = $host;
+				$host_without_www = str_replace( 'www.', '', $host );
```

Comment:
This is too loose in the search and replacement. It must only consider a hostname that starts with `www.`. If the hostname happens to include `www.` elsewhere, it doesn't matter.

Similarly, we only want to remove a leading `www.` if it is present. `str_replace` will replace all occurrences, so we can't use that.

Turns out we already have a method on the base `Module` class for doing this: `Module::permute_site_url` – the difference is that it returns an array of URLs, so we just need to map the return value into host names first, otherwise it does exactly what we need 👍

Reply:
👍